elastic/kibana
# [Cloud Posture] CSPM support - dashboard onboarding per policy template

Resolves #146589

## Summary
This PR is mainly focused on the onboarding process now that we have 2 installations.

Note-worthy features introduced in this PR:
- splits the dashboard into two tabs, `Cloud` and `Kubernetes`
- each tab displays findings based on the `rule.benchmark.id` value. later to be changed to `policy_template` field.
- `policy_template` field was added to the `scores` index
- new initial installation prompt
- per tab installation prompt
- priority order to decide which tab to display
```
     1. cspm in general is preferred
     2. cspm has findings
     3. kspm has findings
     4. cspm is insatlled
     5. kspm is installed 
```

### Initial installation prompt
https://user-images.githubusercontent.com/51442161/210767036-82032ed3-55b9-49bc-a54a-ea5ca1c6353f.mov

### Per tab installation prompt
https://user-images.githubusercontent.com/51442161/210767100-6f71a658-0b4e-4bb3-b151-44a3e0866599.mov

## Debts
- dashboard links to findings, to filter results based on Cloud/Kubernetes dashboard ([task](https://github.com/elastic/kibana/issues/146590))
- dashboard selected tab should be URL based ([task](https://github.com/elastic/kibana/issues/148435))

## Steps to test
- from a clean state, go to csp dashboard/findings page
- you should see a general installation prompt, with links to CSPM and KSPM integration installations forms, both links should work.
- install the **KSPM** integration first. after installation return to the dashboard page
- the dashboard should display 2 tabs now, the dashboard prefers the tab with the installed integration, so you should be on the KSPM tab. if your findings are still on their way, you should see a `no findings yet` prompt.
- after some time findings shout arrive, and all displayed findings in the KSPM dashboard should only be from findings that have `rule.benchmark.id` value of `cis_k8s`. this is temporary until we have the `policy_template` field on the findings.
- note that the dashboard links to the findings page are not yet filtering the results in the findings page, this will be done on the [follow-up task](https://github.com/elastic/kibana/issues/146590)
- go to the CSPM tab, you should see a prompt asking you to install the CSPM integration, the link should send you to the installation form.
- after installation and going back to the dashboard, you will be on the Kubernetes tab since it has findings, click on the Cloud tab, you should see the `no findings yet` prompt. currently, there is no real way to generate those findings. you'll need to manually add a finding with a `rule.benchmark.id: "cis_aws"` field in the dev tools
- going back to the dashboard, you should see that finding in the Cloud tab, and the rest of your findings on the Kubernetes tab.
